pgxn/neon/pagestore_smgr.c

```diff
@@ -1703,10 +1845,10 @@ neon_extend(SMgrRelation reln, ForkNumber forkNum, BlockNumber blkno,
 	 */
 	if (lsn == InvalidXLogRecPtr)
 	{
-		lsn = GetXLogInsertRecPtr();
-		SetLastWrittenLSNForBlock(lsn, reln->smgr_rnode.node, forkNum, blkno);
+		lsn = ProcLastRecPtr;
```

Comment:
I wonder if using ProcLastRecPtr is correct here?
Assume that some other backend B2 has done something with this relation (for example also extend it).
The this backend B1 is calling smgrextend. But please notice that at this moment it has not yet write something to the WAL, so it's  `ProcLastRecPtr`can be far in the past and smaller than LSN of pages appended to the relation by backend B2.